src/p2p/supersocket.erl

```diff
@@ -0,0 +1,35 @@
+%% this module contains a supervised socket server thingy
+-module(supersocket).
+-behaviour(supervisor).
+
+-export([start_socket_server/0]).
+-export([start_socket/0]).
+-export([init/1]).
+
+-include_lib("eunit/include/eunit.hrl").
+-include_lib("stdlib/include/assert.hrl").
+
+start_socket_server() ->
+  supervisor:start_link({local, ?MODULE}, ?MODULE, []).
+
+
+init([]) ->
+  Port = 8384,
+  %% what is {packet, line} idk?
+  {ok, ListenSocket} = gen_tcp:listen(Port, [{active,once}, {packet,line}]),
+  io:fwrite("supersocket listening on port ~p\n", [inet:port(ListenSocket)]),
+  spawn_link(fun empty_listeners/0),
+  {ok, {{simple_one_for_one, 60, 3600},[
+    {socket,
+    {supersocket_tcplistener, start_link, [ListenSocket]}, % pass the socket!
+    temporary, 1000, worker, [sockserv_serv]}
+  ]}}.
+
+start_socket() ->
+  supervisor:start_child(?MODULE, []).
+
+%% Start with 20 listeners so that many multiple connections can join at once
+%% without waiting on previous one to finish connecting.
+empty_listeners() ->
+  [start_socket() || _ <- lists:seq(1,20)],
```

Comment:
is this python?

Reply:
copy pastaed code from learnyousomeerlang is what it is

Reply:
you mean haskell? D: